src/extensions/averagewaveforms/AverageWaveformsView/setupElectrodes.ts

```diff
@@ -112,20 +127,21 @@ const setupElectrodes = (args: {width: number, height: number, electrodeLocation
     if (layoutMode === 'vertical') {
         return setupVerticalElectrodes({width, height, electrodeIds})
     }
+    const electrodeLocations2 = fixDegenerateCase(electrodeLocations)
     const W = width - 10 * 2
     const H = height - 10 * 2
     const canvasAspect = W / H
 
-    const radius = computeRadius(electrodeLocations)
-    let boundingBox = getElectrodesBoundingBox(electrodeLocations, radius)
+    const radius = computeRadius(electrodeLocations2)
+    let boundingBox = getElectrodesBoundingBox(electrodeLocations2, radius)
     let boxAspect = getWidth(boundingBox) / getHeight(boundingBox)
 
-    let realizedElectrodeLocations = electrodeLocations
+    let realizedElectrodeLocations = electrodeLocations2
     if ((boxAspect > 1) !== (canvasAspect > 1)) {
         // if the two aspect ratios' relationship to 1 is different, then one is portrait
         // and the other landscape. We should then correct by rotating the electrode set 90 degrees.
         // note: a 90-degree right rotation in 2d makes x' = y and y' = -x
```

Comment:
A 90-degree right rotation *about the origin*, I meant here.

Reply:
i'll update that